### DISCUSSION TITLE
Pass extra info to NER

### DISCUSSION THREAD

**Discussion by Phat-Loc** [explosion/spaCy#8139] [Help: Coding & Implementations] [answered]
For my use case I use senter on a doc to split the sentences then I apply textcat to each sentence. Finally I apply NER to each of the sentence. Is there a way for me to pass the textcat output into NER so it takes which textcat the sentence is in into account? Or should I just train a separate NER model per textcat and apply NER coresspondingly?

**Comment by polm** [comment, marked as answer]
There's no way to pass extra context to the built-in NER model, so if you want to do this you'd have to train a different NER model per category I guess.

If you do this I would be sure to train an NER model without category information to use as a baseline.

**Reply by adrianeboyd** [reply]
In general, if you create a custom `tok2vec` architecture, you can pass additional features to the `HashEmbed` aside from the ones supported by `doc.to_array()`. Here's an example from a user that uses custom extensions:

https://github.com/explosion/spaCy/discussions/6527

However, for features that come from preceding components in the pipeline, it would be easiest to wait for spaCy v3.1 for this, since it requires some additional updates so that the `ner` component can see the `doc.cats` predicted during training.

For spaCy v3.0 it's not impossible, but it requires a bit more effort. You'd need to split your task up into two separate pipelines during training and write a custom corpus reader in order for this to work with v3.0. Your custom corpus reader would need to make sure that `example.predicted` contains the predicted cats while you're training the the second `ner` pipeline. But I think it should be possible to try out the basic idea if you'd like...
